src/uct/ib/mlx5/dv/ib_mlx5dv_md.c

```diff
@@ -1651,10 +1651,21 @@ static void uct_ib_mlx5_devx_check_odp(uct_ib_mlx5_md_t *md,
         goto no_odp;
     }
 
-    ucs_debug("%s: ODP is supported, version %d",
-              uct_ib_device_name(&md->super.dev), version);
-    md->super.reg_nonblock_mem_types = UCS_BIT(UCS_MEMORY_TYPE_HOST);
     md->super.gva_mem_types          = UCS_BIT(UCS_MEMORY_TYPE_HOST);
+    md->super.reg_nonblock_mem_types = UCS_BIT(UCS_MEMORY_TYPE_HOST);
+
+    if (ucs_arch_get_cpu_model() == UCS_CPU_MODEL_NVIDIA_GRACE) {
+        /* Only supported on systems with coherent memory */
+        md->super.reg_nonblock_mem_types |= UCS_BIT(
+                UCS_MEMORY_TYPE_CUDA_MANAGED);
+    }
```

Comment:
Can we use the configuration passed from UCP level?
And I think it is better to introduce configuration option on UCT level as well. And set the value for grace CPU in ucx.conf.

Reply:
are you referring to using configuration `UCX_REG_NONBLOCK_MEM_TYPES`? if yes, i think this one was to just enforce nonblock flag when doing memory registrations, not to enable odp for cuda-managed.

for UCT we could add `UCX_IB_NONBLOCK_SUPPORT=n`, and add it to ucx.conf is that what you meant?

Reply:
maybe add something like
UCX_IB_REG_NONBLOCK_MEM_TYPES=cuda-managed to ucx.conf for Grace

Reply:
added `UCX_IB_ODP_MEM_TYPES` to differentiate from the UCP layer:
- IB config: which non-block do we want to advertize when ODP is setup
- UCP config: which mem_type we want to always register with nonblock flag